### PR TITLE
Adds support for reading Prometheus config and links ruler with runner 

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
## Description

Fixes: https://github.com/timescale/o11y-team-applications/issues/180

This PR adds support for reading the Prometheus configuration file, validates it, and logs as per content in the file. Based on the details in configuration, it starts the rules-manager. This PR also contains tests for reading and validating Prometheus configuration.

Next PRs will be:
1. Implement the `/api/v1/rules` API
2. E2E tests for Rules-manager

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Review by cevian — APPROVED** [review]
I think our flag names should copy Prometheus naming as much as possible. Approving since I don't want to hold up merge but suggest going over naming with @paulfantom

**Review by antekresic — APPROVED** [review]
Also approved but with caveat to align the flags with Prometheus, like previously mentioned.

File: pkg/rules/testdata/alert_config.good.config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  evaluation_interval: 1m
+
+alerting:
+  alertmanagers:
+  - scheme: https
+    static_configs:
+    - targets: []
+
+rule_files:
+  - rules.yaml
```
Comment (niksajakovljevic) [inline pkg/rules/testdata/alert_config.good.config.yaml:12]: Nit: new line

File: pkg/rules/testdata/no_rules.bad.config.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+global:
+  scrape_interval: 20s
+  evaluation_interval: 30s
+
+rules_files: []
```
Comment (niksajakovljevic) [inline pkg/rules/testdata/no_rules.bad.config.yaml:5]: nit: new line

Comment (niksajakovljevic) [reply pkg/rules/testdata/no_rules.bad.config.yaml:5]: appears in other test files  you've added

**Pull request merged**